### PR TITLE
engine(dm): unfresh sync unit no need to request resource 

### LINKED ISSUE #6784: DMWorkerSync should not request resource after metadata is loaded into checkpoint
### Is your feature request related to a problem?

to provide more freedom for scheduling

### Describe the feature you'd like

as title

### Describe alternatives you've considered

_No response_

### Teachability, Documentation, Adoption, Migration Strategy

_No response_

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve?

Issue Number: close #6784 

### What is changed and how it works?
- if sync is unfresh, no need to request resource.

#### Tests 

 - Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

File: engine/executor/openapi.go [hunk 1]
```diff
@@ -43,13 +45,16 @@ func newJobAPIServer() *jobAPIServer {
 
 func (s *jobAPIServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	if h, ok := s.match(r.URL.Path); ok {
+		log.Debug("serve http request", zap.String("path", r.URL.Path))
```
Comment (lance6716) [inline engine/executor/openapi.go:48]: @gozssky can the framework log OpenAPI request/response soon?

**Review by lance6716 — APPROVED** [review]
if we need to save table checkpoint for `loadTableStructureFromDump` in future, we can save it before saving global checkpoint. Please left a comment to remind us.

File: engine/executor/openapi.go (hunk 1 above)
Comment (gozssky) [reply engine/executor/openapi.go:48]: I think it will be soon. This log can be kept here. If there is any better way, we can improve it later.

**Review by amyangfei — APPROVED** [review]

**Pull request merged**